geopandas/tools/overlay.py

```diff
@@ -361,6 +363,7 @@ def _make_valid(df):
             # respective Multi version)
             dissolved = exploded.dissolve(by="level_0")
             result.loc[is_collection, geom_col] = dissolved[geom_col].values
+            result = result[result.geometry.notna()]
```

Comment:
This might not be needed, because below we still do a filter operation to only keep the geometries of a certain type (for which None will always gives False, and get filtered out)